wallet-new/test/unit/Test/Spec/Addresses.hs

```diff
@@ -129,7 +129,7 @@ prepareAddressFixture n = do
 prepareAddressesFixture
     :: Int  -- ^ Number of Accounts to create.
     -> Int  -- ^ Number of 'Address per account to create.
-    -> Fixture.GenPassiveWalletFixture (M.Map V1.AccountIndex [V1.WalletAddress])
+    -> Fixture.GenPassiveWalletFixture (M.Map V1.AccountIndex [V1.WalletAddress], Int)
```

Comment:
Question: wouldn't be more natural to _not_ modify this function to yield the total, but rather extend `withAddressFixtures` like this?

```
 withAddressFixtures n =
   Fixture.withPassiveWalletFixture $ do
       prepareAddressFixture (max 0 (n - 1))
```

This way, we account for this +1/-1 offset only there, but the rest of the fixture code doesn't even know about that. Plus, we could now invoke our `withAddressFixtures` with the numbers _the tester would expect_, not the numbers _we would expect after we would subtract or add 1 to them_.

Did you try exploring this design trail?

Reply:
@adinapoli-iohk Does this change the logic of the tests? In the sense that now the caller has to specify how many addresses he eventually want to have and not how many new he wants to create.

Reply:
Uhm, I guess you are right. I suspect the fact we are dealing with just integers here doesn't help as seeing calls like `withAddressesFixtures 3 4` tells me nothing about the semantic of those two integers. 

Do you think it might be a good idea to create two `newtype` wrappers so that, for example, we could write:

``` haskell
newtype DesiredNewAddresses = DesiredNewAddrs Int
newtype DesiredNewAccounts   = DesiredNewAccs Int

prepareAddressesFixture
    :: DesiredNewAccounts  -- ^ Number of Accounts to create.
    -> DesiredNewAddresses  -- ^ Number of 'Address per account to create.
    -> Fixture.GenPassiveWalletFixture (M.Map V1.AccountIndex [V1.WalletAddress], Int)
prepareAddressesFixture acn adn = ..

--- and later
withAddressesFixtures (DesiredNewAccs 3) (DesiredNewAddrs 4) $ \_ layer _ (_, total) -> do
[...]
```

Now we have types which tell us what is what. Yes, they make the invocation site bulkier, but at the same time it's less easy to make mistakes and swap the params around. We could follow the some pattern for `withAddressFixture`.

Overkill?

This is orthogonal to my first initial comment (for which I agree perhaps it's better to go with your solution). Sorry for the pick back and forth, but I can see how these tests are important and they also requires careful thinking, so it might be nice spending a bit more time trying to make them more ergonomic 😉